lambdas/issuecredential/src/main/java/uk/gov/di/ipv/cri/kbv/api/builder/CheckDetailsBuilder.java

```diff
@@ -0,0 +1,110 @@
+package uk.gov.di.ipv.cri.kbv.api.builder;
+
+import uk.gov.di.ipv.cri.kbv.api.domain.CheckDetail;
+import uk.gov.di.ipv.cri.kbv.api.domain.KbvQuality;
+import uk.gov.di.ipv.cri.kbv.api.domain.KbvQuestion;
+import uk.gov.di.ipv.cri.kbv.api.domain.KbvQuestionAnswerSummary;
+import uk.gov.di.ipv.cri.kbv.api.domain.QuestionAnswerPair;
+import uk.gov.di.ipv.cri.kbv.api.domain.QuestionState;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Comparator.comparingInt;
+
+public class CheckDetailsBuilder {
+    private static final int UNSUITABLE_QUESTION_QUALITY = 0;
+    private static QuestionState questionState;
+    private static KbvQuestionAnswerSummary answerSummary;
+    private static Map<String, Integer> kbvQualityMapping;
+    private List<List<QuestionAnswerPair>> qaBatches;
+    private Stream<List<QuestionAnswerPair>> batchWithExclusion;
+    private Stream<CheckDetail> checkDetailStream;
+    private Stream<String> qaIds;
+
+    public static CheckDetailsBuilder init(
```

Comment:
You can use a constructor here instead of a static init method. Having this static method reduces its flexibility and readability.

Reply:
How? it just a proxy for   `return new BuilderSteps(questionPairs, answerSummary, kbvQualityMapping)`